### PR TITLE
feat(@clayui/form): add missing hidden input element

### PR DESCRIPTION
**Pull request by julien**

### CODE REVIEW

File: packages/clay-form/src/InputWithMultiSelect.tsx [hunk 1]
```diff
@@ -45,6 +45,11 @@ export interface IProps extends React.HTMLAttributes<HTMLInputElement> {
 	 */
 	items: Array<string>;
 
+	/**
+	 * Value used for the hidden input's name attribute
```
Comment (carloslancha) [inline packages/clay-form/src/InputWithMultiSelect.tsx]: Maybe is better "Value used for every selectedItem hidden input's name attribute"

File: packages/clay-form/src/InputWithMultiSelect.tsx [hunk 2]
```diff
@@ -181,36 +187,39 @@ const ClayMultiSelect: React.FunctionComponent<IProps> = ({
...
+												if (i === items.length - 1) {
+													lastItemRef.current = ref;
+												}
+											}}
+											spritemap={spritemap}
+											tabIndex={0}
+										>
+											{item}
+										</ClayLabel>
+										<input name={inputName}
```
Comment (carloslancha) [inline packages/clay-form/src/InputWithMultiSelect.tsx]: Add blank line

File: packages/clay-form/src/InputWithMultiSelect.tsx [hunk 3]
```diff
@@ -181,36 +187,39 @@ const ClayMultiSelect: React.FunctionComponent<IProps> = ({
...
+											type="hidden"
+											value={item} />
```
Comment (carloslancha) [inline packages/clay-form/src/InputWithMultiSelect.tsx]: `/>` in a new line

File: packages/clay-form/src/__tests__/__snapshots__/InputWithMultiSelect.tsx.snap [hunk 4]
```diff
@@ -44,6 +44,11 @@ exports[`ClayInputWithMultiSelect renders as not valid 1`] = `
         className="form-control form-control-tag-group"
         onKeyDown={[Function]}
       >
+        <input
+          name=""
```
Comment (wincent) [inline packages/clay-form/src/__tests__/__snapshots__/InputWithMultiSelect.tsx.snap]: I wanted to double check what would happen when using the component's default (empty string) name:

> If no name is specified, or name is empty, the input's value is not submitted with the form.

([Source](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input))

So seems ok.

File: packages/clay-form/src/__tests__/InputWithMultiSelect.tsx [hunk 5]
```diff
@@ -128,7 +128,7 @@ describe('Interactions', () => {
 		expect(document.activeElement!.tagName).toEqual('BODY');
 
 		fireEvent.keyDown(
-			container.querySelector('input') as HTMLInputElement,
+			container.querySelector('input.form-control-inset') as HTMLInputElement,
```
Comment (wincent) [inline packages/clay-form/src/__tests__/InputWithMultiSelect.tsx]: This smells a bit implementation-detailish. Might be better to target by type eg.

```javascript
container.querySelector('input[type=text]')
```

File: packages/clay-form/src/__tests__/__snapshots__/InputWithMultiSelect.tsx.snap (hunk 4 above)
Comment (julien) [reply packages/clay-form/src/__tests__/__snapshots__/InputWithMultiSelect.tsx.snap]: Thanks for checking that out.

**Comment by matuzalemsteles** [comment]
Fixes #2379

**Pull request merged**